parameter/criterion/include/criterion/Criterion.h

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2011-2015, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#pragma once
+
+#include "client/CriterionInterface.h"
+#include "XmlSource.h"
+#include <log/Logger.h>
+
+#include <map>
+#include <string>
+#include <functional>
+#include <stdexcept>
+
+namespace core
+{
+namespace selection
+{
+namespace criterion
+{
+
+/** Criterion object used to apply rules based on system state */
+class Criterion : public IXmlSource, public CriterionInterface
+{
+public:
+    using InvalidCriterionError = std::runtime_error;
+
+    /** @param[in] name the criterion name
+     *  @param[in] logger the main application logger
+     */
+    Criterion(const std::string& name, const Values& values, core::log::Logger& logger);
+
+    // @{
+    /** @see CriterionInterface */
```

Comment:
If you want this description to be set for the whole group of methods, the correct way is: http://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdaddtogroup